units/src/amount/unsigned.rs

```diff
@@ -219,7 +219,7 @@ impl Amount {
     ///
     /// Be aware that integer division loses the remainder if no exact division
     /// can be made.  This method rounds up ensuring the transaction fee-rate is
-    /// sufficient.  If you wish to round-down, use the unchecked version instead.
+    /// sufficient.
```

Comment:
In be5ec14d5d0f859d35e6c8f243c8cf501d2c6835:

It should say "If you wish to round down, use the `/` operator instead.". (And if possible, we should put symmetric docs on the `/` operator saying that this does truncating division, which is likely the wrong thing for a user dividing an amount by a weight, and that they should consider using `checked_div_by_weight` instead.)

Reply:
We don't implement `ops::Div<Weight> for Amount` - are you suggesting that we add it?

Reply:
Hmm, if we don't have it let's not add it for now.

I would like to have it, but the discrepancy in rounding behavior would be a bit weird so maybe we want to discuss it. So ok, for now your original text is fine.

Reply:
ugh, my bad. its in `fee_rate` module.

Reply:
Do you remember was there  a reason we didn't use `div_by_weight_ceil` and `div_by_weight_floor` (and not implement `ops::Div` at all)?

Reply:
After commenting the current code, it really begs the question why we implemented it at all?

```rust
impl Div<Weight> for Amount {
    type Output = FeeRate;

    /// Truncating integer division.
    ///
    /// This is likely the wrong thing for a user dividing an amount by a weight, consider using
    /// `checked_div_by_weight` instead.
    fn div(self, rhs: Weight) -> Self::Output { FeeRate(self.to_sat() * 1000 / rhs.to_wu()) }
}
```

Reply:
Moving discussion to https://github.com/rust-bitcoin/rust-bitcoin/issues/3563

Reply:
> After commenting the current code, it really begs the question why we implemented it at all?

Well the `checked_div_by_weight` was added after if you're referring to the comment directing people to used the `checked_` version and not to use this version.

Reply:
>it really begs the question why we implemented it at all?

Because it makes mathematical sense to have.

But given that it's kinda a footgun maybe we should drop it..

Reply:
Dropping it works for me if there's checked_ceil and checked_floor version.  Should probably be deprecation messages so people have a clear upgrade path.  Although the easiest thing is to just leave it since it's already there.

Reply:
Let's just keep it. I think its behavior is unsurprising, even if it's usually the wrong thing to do.

Reply:
Yeah, I think it could be revisited in the future

Reply:
Cool, cheers. FTR consensus in this thread is fully covered by this comment on the issue: https://github.com/rust-bitcoin/rust-bitcoin/issues/3563#issuecomment-2452726869